pandas-stubs/core/indexes/interval.pyi

```diff
@@ -47,37 +69,57 @@ class IntervalIndex(IntervalMixin, ExtensionIndex):
     @classmethod
     def from_tuples(
         cls,
-        data,
+        data: Sequence[tuple[int, int]]
+        | Sequence[tuple[float, float]]
+        | Sequence[tuple[DatetimeLike, DatetimeLike]]
+        | npt.NDArray,
         closed: IntervalClosedType = ...,
         name: Hashable = ...,
         copy: bool = ...,
         dtype: IntervalDtype | None = ...,
     ) -> IntervalIndex: ...
+    def __contains__(self, key: Any) -> bool: ...
     def astype(self, dtype: DtypeArg, copy: bool = ...) -> IntervalIndex: ...
     @property
     def inferred_type(self) -> str: ...
     def memory_usage(self, deep: bool = ...) -> int: ...
+    def is_monotonic_decreasing(self) -> bool: ...
+    def is_unique(self) -> bool: ...
     @property
     def is_overlapping(self) -> bool: ...
-    def get_loc(self, key, tolerance=...) -> int | slice | np.ndarray: ...
+    # Note: tolerance removed as it has no effect
+    def get_loc(
+        self,
+        key: Label,
+        method: str | None = ...,
+    ) -> int | slice | np.ndarray: ...
     def get_indexer(
         self,
-        targetArrayLike,
+        target: Index,
         method: str | None = ...,
         limit: int | None = ...,
         tolerance=...,
     ) -> np.ndarray: ...
     def get_indexer_non_unique(
-        self, targetArrayLike
-    ) -> tuple[np.ndarray, np.ndarray]: ...
+        self, target: Index
+    ) -> tuple[npt.NDArray[np.intp], npt.NDArray[np.intp]]: ...
+    @property
+    def left(self) -> Index: ...
+    @property
+    def right(self) -> Index: ...
+    @property
+    def mid(self) -> Index: ...
+    @property
+    def length(self) -> Index: ...
     def get_value(self, series: ABCSeries, key): ...
     @property
     def is_all_dates(self) -> bool: ...
-    def __lt__(self, other): ...
-    def __le__(self, other): ...
-    def __gt__(self, other): ...
-    def __ge__(self, other): ...
 
 def interval_range(
-    start=..., end=..., periods=..., freq=..., name=..., closed: str = ...
-): ...
+    start: int | float | DatetimeLike | None = ...,
+    end: int | float | DatetimeLike | None = ...,
+    periods: int | None = ...,
+    freq: int | str | BaseOffset | None = ...,
```

Comment:
docs say `DateOffset` here, not `BaseOffset`

Reply:
Fixed.